File: java/724.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;
        for(int num:nums){
            sum += num;
        }
        int tmp = 0;
        for(int i = 0;i < nums.length;i++){
            if(tmp * 2 + nums[i] == sum){
                return i;
            }
            tmp += nums[i];
        }
        return -1;
    }
}

File: go/724.go
func pivotIndex(nums []int) int {
	sum := 0
	for _, v := range nums {
		sum += v
	}
	left := 0
	for i, v := range nums {
		if left*2+v == sum {
			return i
		}
		left += v
	}
	return -1
}